### PR TITLE
Cart functionality/sone

### PR DESCRIPTION
**Pull request by SoneThaya**

### CODE REVIEW

File: client/src/state/actions/cartActions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    payload: {
+      product: data.id,
+      name: data.product_name,
+      image: data.product_img,
+      price: data.product_price,
+      quantity,
+    },
+  });
+
+  localStorage.setItem("cartItems", JSON.stringify(getState().cart.cartItems));
```
Comment (anthonyamaro15) [inline client/src/state/actions/cartActions.js]: I'm a bit confused about why you are saving cart data in localStorage?

File: client/src/store.js [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
 const reducer = combineReducers({
   userLogin: userLoginReducer,
-  userRegister: userRegisterReducer
+  userRegister: userRegisterReducer,
+  cart: cartReducer,
 });
 
+const cartItemsFromStorage = localStorage.getItem("cartItems")
+  ? JSON.parse(localStorage.getItem("cartItems"))
+  : [];
```
Comment (anthonyamaro15) [inline client/src/store.js]: I don't think you need to save this data to localStorage you can just get it from the backend whenever a user log in

**Review by anthonyamaro15 — CHANGES_REQUESTED** [review]
I left a few comments

**Comment by SoneThaya** [comment]
edited the code, will create a new pull request

**Pull request closed without merging**